src/resdk/resources/base.py

```diff
@@ -132,6 +132,9 @@ def assert_fields_unchanged(field_names):
                     payload[field_name] = self._dehydrate_resources(
                         getattr(self, field_name)
                     )
+            if "sample" in payload:
+                payload["entity"] = copy.deepcopy(payload["sample"])
```

Comment:
Why do you need a deep copy? The original reference (i.e., `payload["sample"]`) is deleted anyway so you can use the same object. 

You can use `payload.pop("sample")` instead of `payload["sample"]` + `del payload["sample"]`

Reply:
Is the same thing also needed in the `else` below? It would be nicer to move this into the `Data.save` method and handle it before this method is called.

Reply:
> Is the same thing also needed in the else below?
Yes it is, thanks!

> It would be nicer to move this into the Data.save method and handle it before this method is called.
True, but to achieve this, one would need to mess with `self._original_values`, which is kinda against what the "original values" should be. The "proper" way is to refactor save method to enable better subclassing but i hoped i could get away without that :-D

Reply:
I see it would be much more challenging then I thought. 

Where does the `sample` name come from? I think that the right solution would be to use `entity` everywhere in the codebase and add `sample` property to the `Data` class as an alisas for `entity`. Would that be feasible?